log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -24,25 +26,50 @@
 class BufferedHandler
         extends Handler
 {
-    private static final int MAX_BATCH_COUNT = 1024;
+    private static final int MAX_BATCH_COUNT = 10240;
     private static final byte[] POISON_MESSAGE = new byte[0];
 
     private final MessageOutput messageOutput;
     private final Thread thread;
     private final BlockingQueue<byte[]> queue = new ArrayBlockingQueue<>(MAX_BATCH_COUNT);
     private final AtomicBoolean closed = new AtomicBoolean();
     private final AtomicLong droppedMessages = new AtomicLong(0);
+    private final PrintStream stdErr;
 
-    public BufferedHandler(MessageOutput messageOutput, Formatter formatter)
+    public BufferedHandler(MessageOutput messageOutput, Formatter formatter, PrintStream stdErr)
     {
         this.messageOutput = requireNonNull(messageOutput, "messageOutput is null");
+        this.stdErr = requireNonNull(stdErr, "stdErr is null");
         setFormatter(requireNonNull(formatter, "formatter is null"));
 
+        setErrorManager(new BufferedHandlerErrorManager());
+
         thread = new Thread(this::logging);
         thread.setName("log-writer");
         thread.setDaemon(true);
     }
 
+    private class BufferedHandlerErrorManager
+            extends ErrorManager
+    {
+        private final AtomicBoolean reported = new AtomicBoolean();
+
+        public synchronized void error(String msg, Exception ex, int code)
+        {
+            if (!reported.compareAndSet(false, true)) {
```

Comment:
No need for an AtomicBoolean. This runs in a syncrhonized block, and no one other than this method accesses `reported`.

Reply:
Also, I'm not sure I understand the logic here. Once this method gets called, it will forever after short-circuit and exit early. What's the purpose of this `BufferedHandlerErrorManager`?